src/GoTrueClient.ts

```diff
@@ -1026,28 +1052,10 @@ export default class GoTrueClient {
     return `${this.url}/authorize?${urlParams.join('&')}`
   }
 
-  private async _unenroll(params: MFAUnenrollParams) {
-    const { data, error } = await this.getUser()
-    if (error) throw error
-    const user: User = data.user
-
-    try {
-      return await _request(
-        this.fetch,
-        'DELETE',
-        `${this.url}/user/${user.id}/factor/${params.factorId}`,
-        {
-          body: { code: params.code },
-          headers: this.headers,
-        }
-      )
-    } catch (error) {
-      if (isAuthError(error)) {
-        return { data: null, error }
-      }
-
-      throw error
-    }
+  private async _unenroll(params: MFAUnenrollParams): Promise<AuthMFAUnenrollResponse> {
+    return await _request(this.fetch, 'DELETE', `${this.url}/factor/${params.factorId}`, {
+      headers: this.headers,
```

Comment:
don't we still need to pass in the code if we want to unenroll a verified factor?
```suggestion
      headers: this.headers,
      body: { code: params.code },
```

Reply:
Ideally not, see the discussion on Slack

Reply:
wait, I think as of right now we still need to pass in a code if we want to unenroll a verified factor. Should we be removing that? But let's continue on Slack yes